Userland/Libraries/LibJS/Runtime/ArrayPrototype.cpp

```diff
@@ -680,26 +680,63 @@ JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::reduce_right)
 // 23.1.3.23 Array.prototype.reverse ( ), https://tc39.es/ecma262/#sec-array.prototype.reverse
 JS_DEFINE_NATIVE_FUNCTION(ArrayPrototype::reverse)
 {
-    auto* array = Array::typed_this(vm, global_object);
-    if (!array)
+    auto* this_object = vm.this_value(global_object).to_object(global_object);
+    if (!this_object)
+        return {};
+    auto length = length_of_array_like(global_object, *this_object);
+    if (vm.exception())
         return {};
 
-    if (array->indexed_properties().is_empty())
-        return array;
+    // do we need an explicit floor here?
```

Comment:
Please don't add comments like this. To whom is this question addressed? What does the spec say?

Reply:
This was just a comment I left since I wanted to ask about this since length is a `size_t`.
I just removed the floor although the spec does use it, is that ok?